Assignment7 Sheqi.out

```diff
@@ -0,0 +1,42 @@
+Assignment 7  Sheqi Zhang
+
+1. Explain the difference between ++*p, *p++ and *++p, if there is any. 
+Answer:
+++*p increments the value at location p points to, then evaluates to incremented value.
+*p++ evaluates the value at location p points to, then advances p.
+*++p increments the value of p, then evaluate the location ++p points to.
+
+2. Is the left to right or right to left order guaranteed for operator precedence?
+Answer:
+In most situations, the left to right order is guarenteed, while for operators like "+=","-=", the order is right to left.
+
+3. What is the advantage of using pointers?
+Answer:
+The pointer provide a way for functions to modify their calling argument, and it supports the dynamic memory allocation. It's more effecient comparing with arrays.
+
+4. 
+ 4.1 "abc" <br>
+	char
+ 4.2 "xyz"[1] - ’y’ <br>
+	Invalid. "xyz"[1] is an array, while 'y' is character;
+ 4.3 ’\0’ == 0 <br>
+	boolean. It is judging if it's wrong.
```

Comment:
right, but which value?  '\0' is just a NULL terminator and by definition, NULL is equal to 0. So this would evaluate as true, which is 1 in C.